### PR TITLE
Make static extensions list configurable

### PR DESCRIPTION
**Pull request by solenko**

### CODE REVIEW

**Comment by eliotsykes** [comment]
Thanks @solenko for a great contribution! Taking a look now.

File: README.md [hunk 1]
```diff
@@ -85,6 +85,21 @@ Any files given the `max_age_fallback` would have the following `Cache-Control`
 Cache-Control: public, max-age=600
 ```
 
+### Configuration
+
+### static_extensions
+
+rack-zippy handle only files with whitelisted extensions. Default extensions list:
```
Comment (eliotsykes) [inline README.md]: Very minor typo: handle -> handles

File: lib/rack-zippy.rb [hunk 2]
```diff
@@ -1,19 +1,26 @@
...
 
     class AssetServer
 
-      # Font extensions: woff, woff2, ttf, eot, otf
-      STATIC_EXTENSION_REGEX = /\.(?:css|js|html|htm|txt|ico|png|jpg|jpeg|gif|pdf|svg|zip|gz|eps|psd|ai|woff|woff2|ttf|eot|otf|swf)\z/i
-
       HTTP_STATUS_CODE_OK = 200
 
+      class << self
+        def static_extension_regex
```
Comment (eliotsykes) [inline lib/rack-zippy.rb]: What do you think about removing the regex completely?

File: test/serveable_file_test.rb [hunk 3]
```diff
@@ -467,6 +467,12 @@ def test_has_static_extension_returns_true_for_flash
                "Should handle flash .swf files"
       end
 
+      def test_has_static_extension_returns_true_for_configured_extension
+        Rack::Zippy.static_extensions << 'csv'
```
Comment (eliotsykes) [inline test/serveable_file_test.rb]: What do you think about resetting the static extensions to their original value in the teardown?

**Comment by eliotsykes** [comment]
This PR would be a strong reason for a 3.0 release :+1: . (I believe there are quite a few rack-zippy installs on version 2.0.x relying on patching the old regex constant.)

**Comment by solenko** [comment]
@eliotsykes, can you please take a look on PR one more time?
All your comments taken in account.

File: test/serveable_file_test.rb [hunk 4]
```diff
@@ -14,6 +14,7 @@ def setup
 
       def teardown
         revert_to_original_working_directory
+        Rack::Zippy.reset_static_extensions
```
Comment (eliotsykes) [inline test/serveable_file_test.rb]: I *think* the reset won't work as intended as the original default array is changed by `Rack::Zippy.static_extensions << 'csv'`.

**Comment by eliotsykes** [comment]
> @eliotsykes, can you please take a look on PR one more time? All your comments taken in account.

Brilliant, thank you! Comment left inline about reset.

File: test/configuration_test.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+      def test_define_setting_create_reset_method
+        ConfiguredClass.define_setting setting_name
+        assert ConfiguredClass.respond_to? "reset_#{setting_name}"
+      end
+
+      def test_reset_setting_to_default_value
+        ConfiguredClass.define_setting 'test_setting', 'default_value'
+        ConfiguredClass.test_setting = 'new_value'
+        ConfiguredClass.reset_test_setting
+        assert ConfiguredClass.test_setting == 'default_value'
```
Comment (eliotsykes) [inline test/configuration_test.rb]: Consider using `assert_equal` on lines 40 and 46

File: test/serveable_file_test.rb (hunk 4 above)
Comment (solenko) [inline test/serveable_file_test.rb]: You are absolutely right. Fixed in b347856.

**Comment by eliotsykes** [comment]
I'm so grateful for the contribution. I aim to release v3.0 as a result of your work. If things go to plan that'll be this weekend. Thank you @solenko :+1:

**Comment by eliotsykes** [comment]
Released in 3.0.0 - thanks @solenko, wonderful contribution.

**Pull request merged**